app/src/main/java/org/mozilla/fenix/home/collections/TabInCollection.kt

```diff
@@ -0,0 +1,262 @@
+ * @param onClick Invoked when the user click on the tab.
+ * @param onRemove Invoked when the user removes the tab informing also if the tab was swiped to be removed.
+ */
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+fun TabInCollection(
+    tab: Tab,
+    isLastInCollection: Boolean,
+    onClick: () -> Unit,
+    onRemove: (Boolean) -> Unit,
+) {
+    val dismissState = rememberDismissState()
+
+    if (dismissState.isDismissed(StartToEnd) || dismissState.isDismissed(EndToStart)) {
+        onRemove(true)
+    }
+
+    SwipeToDismiss(
+        state = dismissState,
+        background = {
+            DismissedTabBackground(
+                dismissDirection = dismissState.dismissDirection,
+                isLastInCollection = isLastInCollection,
+            )
+        }
+    ) {
+        // We need to clip the top bounds to avoid this item drawing shadows over the above item.
+        // But we need to add this shadows back to have a clearer separation between tabs
+        // when one is being swiped away.
+        val clippingModifier by remember {
+            derivedStateOf {
+                try {
+                    if (dismissState.progress.fraction != 1f) Modifier else Modifier.clipTop()
+                } catch (e: NoSuchElementException) {
+                    // `androidx.compose.material.Swipeable.findBounds` couldn't find anchors.
+                    // Happened once in testing when deleting a tab. Could not reproduce afterwards.
+                    Modifier.clipTop()
+                }
+            }
+        }
+
+        Card(
+            modifier = clippingModifier
+                .fillMaxWidth(),
+            shape = if (isLastInCollection) bottomTabShape else middleTabShape,
+            backgroundColor = FirefoxTheme.colors.layer2,
+            elevation = 5.dp,
+        ) {
+            FaviconListItem(
+                label = tab.title,
+                description = shortenUrl(tab.url),
+                onClick = onClick,
+                url = tab.url,
+                iconPainter = painterResource(drawable.ic_close),
+                iconDescription = stringResource(string.remove_tab_from_collection),
+                onIconClick = { onRemove(false) },
+            )
+        }
+    }
+}
+
+/**
+ * Composable used to display the background of a [Tab] shown in collections that is being swiped left or right.
+ *
+ * @param dismissDirection [DismissDirection] of the tab being swiped depending on which this composable
+ * will also indicate the swipe direction by placing a warning icon at the start of the swipe gesture.
+ * If `null` the warning icon will be shown at both ends.
+ * @param isLastInCollection Whether the tab is to be shown between others or as the last one in collection.
+ */
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+private fun DismissedTabBackground(
+    dismissDirection: DismissDirection?,
+    isLastInCollection: Boolean,
```

Comment:
I also have similar feelings for `TabInCollection`

Reply:
Seems like the decision is about who should control what shape the background looks like for a dismissed tab.

With all the other values related to size, colors and imagery controlled from this composable and no way for it to prevent clients passing a circle for example and breaking it seems like it is better for this composable to be in full control of the shape also and for it to be left to contain all rules on how to adapt a bit of state to show this particular piece of UI.
If others want to pass another shape probably the usecase is different and other properties (sizes, colors, imagery) have to be updated also.